static/js/realm_logo.js

```diff
@@ -23,39 +25,35 @@ var realm_logo = (function () {
 
         return upload_widget.build_direct_upload_widget(
             get_file_input,
-            $("#realm_logo_file_input_error").expectOne(),
-            $("#realm_logo_upload_button").expectOne(),
-            upload_function,
+            args.file_input_error_elem.expectOne(),
+            args.upload_button_elem.expectOne(),
+            args.upload_function,
             page_params.max_logo_file_size
         );
     };
 
-    exports.build_realm_night_logo_widget = function (upload_function) {
-        var get_file_input = function () {
-            return $('#realm_night_logo_file_input').expectOne();
+    exports.build_realm_day_mode_logo_widget = function (upload_function) {
+        var args = {
+            delete_button_elem: $("#realm_logo_delete_button"),
+            file_input_elem: $('#realm_logo_file_input'),
+            file_input_error_elem: $("#realm_logo_file_input_error"),
+            is_night_mode: false,
+            upload_button_elem: $("#realm_logo_upload_button"),
+            upload_function: upload_function,
         };
-        if (page_params.realm_night_logo_source === 'D') {
-            $("#realm_night_logo_delete_button").hide();
-        } else {
-            $("#realm_night_logo_delete_button").show();
-        }
-        var data = {night: JSON.stringify(true)};
-        $("#realm_night_logo_delete_button").on('click', function (e) {
-            e.preventDefault();
-            e.stopPropagation();
-            channel.del({
-                url: '/json/realm/logo',
-                data: data,
-            });
-        });
+        return build_realm_logo_widget(args);
+    };
 
-        return upload_widget.build_direct_upload_widget(
-            get_file_input,
-            $("#realm_night_logo_file_input_error").expectOne(),
-            $("#realm_night_logo_upload_button").expectOne(),
-            upload_function,
-            page_params.max_logo_file_size
-        );
+    exports.build_realm_night_mode_logo_widget = function (upload_function) {
+        var args = {
+            delete_button_elem: $("#realm_night_logo_delete_button"),
+            file_input_elem: $('#realm_night_logo_file_input'),
+            file_input_error_elem: $("#realm_night_logo_file_input_error"),
+            is_night_mode: true,
+            upload_button_elem: $("#realm_night_logo_upload_button"),
+            upload_function: upload_function,
+        };
+        return build_realm_logo_widget(args);
```

Comment:
See my commentary on https://github.com/zulip/zulip/pull/11688.  I think this is a great direction, but in the ideal case we can remove most of these args by using classes to find the elements in question.  

@sahil839 FYI